internal/pkg/githubapi/promotion.go

```diff
@@ -222,6 +235,12 @@ func GeneratePromotionPlan(ghPrClientDetails GhPrClientDetails, config *cfg.Conf
 			}
 		}
 	}
-
 	return promotions, nil
 }
+
+func GeneratePromotionPlan(ghPrClientDetails GhPrClientDetails, config *cfg.Config, configBranch string) (map[string]PromotionInstance, error) {
+	// TODO refactor tests to use the two functions below instead of this one
```

Comment:
Yes.
It wouldn't improve the coverage but it would help you us understand where a failure is and make the tests a bit simpler and easier to extend.